cyclops/models/wrappers/pt_model.py

```diff
@@ -881,94 +903,96 @@ def predict_proba(self, X, **predict_params):
                 by default None
         only_predictions : bool, optional
             Whether to return only the predictions rather than the dataset \
-                with predictions, by default False
+                with predictions when the input is a Hugging Face Datset, \
+                by default False
 
         Returns
         -------
-        Union[Dataset, DatasetColumn]
+        Union[Dataset, DatasetColumn, np.ndarray]
             Dataset containing the predictions or the predictions array.
 
-        """
-        if model_name:
-            pred_column = f"{prediction_column_prefix}.{model_name}"
-        else:
-            pred_column = f"{prediction_column_prefix}.{self.model_.__class__.__name__}"
-
-        if isinstance(feature_columns, str):
-            feature_columns = [feature_columns]
-
-        def get_predictions(examples):
-            stacked = []
-            for feature in feature_columns:
-                stacked.append(torch.stack(examples[feature]))
-            X = torch.cat(stacked, dim=1)
-            return {pred_column: self.predict(X)}
-
-        format_kwargs = {} if transforms is None else {"transform": transforms}
-        with dataset.formatted_as(
-            "custom" if transforms is not None else "torch",
-            columns=feature_columns,
-            **format_kwargs,
-        ):
-            preds_ds = dataset.map(
-                get_predictions,
-                batched=True,
-                batch_size=self.batch_size,
-                remove_columns=dataset.column_names,
-            )
+        Raises
+        ------
+        ValueError
+            If `X` is a Hugging Face Dataset and the feature column(s) is not provided.
 
-            if only_predictions:
-                return DatasetColumn(preds_ds.with_format("numpy"), pred_column)
-            dataset = concatenate_datasets([dataset, preds_ds], axis=1)
-            return dataset
+        """
+        # Input is a Hugging Face Dataset
+        if isinstance(X, Dataset):
+            if feature_columns is None:
+                raise ValueError(
+                    "Missing feature columns 'feature_columns'. Please provide \
+                    the name of feature columns when using \
+                    a Hugging Face dataset as the input."
+                )
+            if isinstance(feature_columns, str):
+                feature_columns = [feature_columns]
+
+            if model_name:
+                pred_column = f"{prediction_column_prefix}.{model_name}"
+            else:
+                pred_column = (
+                    f"{prediction_column_prefix}.{self.model_.__class__.__name__}"
+                )
+
+            def get_predictions(examples):
+                stacked = []
+                for feature in feature_columns:
+                    stacked.append(torch.stack(examples[feature]))
+                X = torch.cat(stacked, dim=1)
+                return {pred_column: self.predict(X)}
```

Comment:
Instead of doing a recursive call here, I propose calling `_evaluation_step` directly in addition to applying the activation function. That is:
```python
output = self._evaluation_step(batch, training=False, **predict_params)
output = self.activation_(output)
```
That way, a dataloader will not be created for every call to `predict_proba`.